internal/index_read.go

```diff
@@ -156,6 +150,13 @@ func readStats(r io.Reader, typ string) (*ReferenceStats, error) {
 	return &stats, nil
 }
 
+func min(a, b int) int {
```

Comment:
The convention is that helper appear after their use, or if shared in a collection of helpers. Please place this below readIntervals.